nuclear_simulator/simulator/core/sim.py

```
@@ -0,0 +1,945 @@
+            "reactivity_components": primary_result['reactivity_components'],
+        }
+        
+        # Update maintenance system if available
+        if hasattr(self, 'maintenance_system') and self.maintenance_system is not None:
+            try:
+                maintenance_work_orders = self.maintenance_system.update(elapsed_minutes, self.dt)
+                # Add maintenance info to result if work orders were created/executed
+                if maintenance_work_orders:
+                    info['maintenance_work_orders'] = [wo.to_dict() for wo in maintenance_work_orders]
+            except Exception as e:
+                warnings.warn(f"Maintenance system update failed: {e}")
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)